test/url_test.rb

```diff
@@ -407,6 +407,13 @@ def test_basic_auth_user_without_password
       assert_equal "Basic #{Base64.encode64("a b").rstrip}", u.basic_auth
     end
 
+    def test_basic_auth?
+      url_with_user = Url.parse("https://a@example.com")
+      assert_equal url_with_user.basic_auth?, true
+      url_without_user = Url.parse("https://example.com")
+      assert_equal url_without_user.basic_auth?, false
```

Comment:
Expectation first :)  `assert_equal false, url_without_user.basic_auth?`.

You could condense it like this too:

```ruby
assert Url.parse("https://a@example.com").basic_auth?
assert !Url.parse("https://example.com").basic_auth?
```